packages/xchain-ethereum/src/client.ts

```diff
@@ -227,64 +270,168 @@ export default class Client implements EthereumClient {
-    if (address && !Client.validateAddress(address)) {
-      return Promise.reject('Invalid Address')
-    } else {
-      const transactions = this._etherscan.getHistory(address)
-      return transactions
+  getTransactions = async (params?: TxHistoryParams): Promise<TxsPage> => {
+    const address = params?.address ?? this.getAddress()
+    const limit = params?.limit ?? 10
+    const offset = params?.offset ?? 0
+
+    let totalCount = 0
+    const transactions: Tx[] = []
+    try {
+      //Calling getAddress without limit/offset to get total count
+      const dAddr = await blockChair.getAddress(this.blockchairNodeUrl, address, this.blockchairNodeApiKey)
+      totalCount = dAddr[address].transactions.length
+
+      const dashboardAddress = await blockChair.getAddress(
+        this.blockchairNodeUrl,
+        address,
+        this.blockchairNodeApiKey,
+        limit,
+        offset,
+      )
+      const txList = dashboardAddress[address].transactions
+
+      for (const hash of txList) {
+        const rawTx = (await blockChair.getTx(this.blockchairNodeUrl, hash, this.blockchairNodeApiKey))[hash]
+        const tx: Tx = {
+          asset: AssetETH,
+          from: rawTx.inputs.map((i: TxIO) => ({ from: i.recipient, amount: baseAmount(i.value, ETH_DECIMAL) })),
+          to: rawTx.outputs
+            // ignore tx with type 'nulldata'
+            .filter((i: TxIO) => i.type !== 'nulldata')
+            .map((i: TxIO) => ({ to: i.recipient, amount: baseAmount(i.value, ETH_DECIMAL) })),
+          date: new Date(`${rawTx.transaction.time} UTC`), //blockchair api doesn't append UTC so need to put that manually
+          type: 'transfer',
+          hash: rawTx.transaction.hash,
+        }
+        transactions.push(tx)
+      }
+    } catch (error) {
+      return Promise.reject(error)
+    }
+
+    return {
+      total: totalCount,
+      txs: transactions,
+    }
+  }
+
+  getTransactionData = async (txId: string): Promise<Tx> => {
+    try {
+      const rawTx = (await blockChair.getTx(this.blockchairNodeUrl, txId, this.blockchairNodeApiKey))[txId]
+      return {
+        asset: AssetETH,
+        from: rawTx.inputs.map((i) => ({ from: i.recipient, amount: baseAmount(i.value, ETH_DECIMAL) })),
+        to: rawTx.outputs.map((i) => ({ to: i.recipient, amount: baseAmount(i.value, ETH_DECIMAL) })),
+        date: new Date(`${rawTx.transaction.time} UTC`), //blockchair api doesn't append UTC so need to put that manually
+        type: 'transfer',
+        hash: rawTx.transaction.hash,
+      }
+    } catch (error) {
+      return Promise.reject(error)
+    }
+  }
+
+  transfer = async ({ memo, amount: amountParam, recipient }: TxParams): Promise<TxHash> => {
+    const amount = baseToAsset(amountParam).amount().toString()
+    const res = await (memo
+      ? this.vaultTx(recipient, amount, memo)
+      : this.normalTx({
+          addressTo: recipient,
+          amount,
+        }))
+
+    return res.hash
```

Comment:
Handle possible cases of errors before returning `res.hash`. `valultTx` / `normalTx` might fail...